Add test_develop configuration

ehelms opened the pull request.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_develop.yaml:
```diff
@@ -0,0 +1,50 @@
+      - github
+      - pollscm: '*/5 * * * *'
+      - schedule-failed-builds: '* * * * *'
+    axes:
```
There's a missing slave axis, with label "fast".  (These are slaves with SSDs etc., use them for core Foreman tests, especially integration tests.)

domcleal commented:
It's missing Xvfb, which adds the following element to `<matrix-project>` in the original XML:

<pre>
  <buildWrappers>
    <org.jenkinsci.plugins.xvfb.XvfbBuildWrapper plugin="xvfb@1.0.16">
      <installationName>system</installationName>
      <screen>1024x768x24</screen>
      <debug>false</debug>
      <timeout>0</timeout>
      <displayNameOffset>1</displayNameOffset>
      <additionalOptions></additionalOptions>
      <shutdownWithBuild>false</shutdownWithBuild>
      <autoDisplayName>false</autoDisplayName>
      <parallelBuild>false</parallelBuild>
    </org.jenkinsci.plugins.xvfb.XvfbBuildWrapper>
  </buildWrappers>
</pre>

We don't specify any of those options actually, it's simply a tickbox in the job configuration.  It'll need adding to JJB (https://github.com/theforeman/jenkins-job-builder/commits/local).

ehelms commented:
I did some updates to this PR to breakout some components into more generic re-usable pieces. 

As for xvfb, looks like upstream jenkins_job_builder supports it (https://github.com/openstack-infra/jenkins-job-builder/blob/master/jenkins_jobs/modules/wrappers.py#L1503); what is the process for updating our copy of jenkins_job_builder to pull in upstream updates?

domcleal commented:
> As for xvfb, looks like upstream jenkins_job_builder supports it (https://github.com/openstack-infra/jenkins-job-builder/blob/master/jenkins_jobs/modules/wrappers.py#L1503); what is the process for updating our copy of jenkins_job_builder to pull in upstream updates?

Ah nice.  I guess a PR against the repo would suffice, though on this occasion I've just merged and pushed it.  I've also submitted our last local change back to OpenStack so we might be able to drop it in future.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/triggers/github_daily.yaml:
```diff
@@ -0,0 +1,6 @@
+- trigger:
+    name: github_daily 
```
Why the _daily suffix?  The poll is every 5 mins.

ehelms replied:
I misread the cron entry - will be updated with next push.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/publishers/release_nightly_build_foreman.yaml:
```diff
@@ -0,0 +1,10 @@
+          - project: release_nightly_build_rpm, release_nightly_build_deb
+            condition: SUCCESS
+            predefined-parameters: |
+              project=foreman-proxy
```
This needs substituting for test_develop.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_develop.yaml:
```diff
@@ -0,0 +1,41 @@
+      - ircbot_freenode
+      - archive:
+          artifacts: 'pkg/*,Gemfile.lock'
+          only-successful: false
```
Change this to `only-if-success` now, the newer JJB had a different name for the setting to our local change.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_proxy_develop.yaml:
```diff
@@ -37,17 +35,11 @@
+          results: 'jenkins/reports/unit/*.xml'
+      - archive:
+          artifacts: 'pkg/*'
+          only-successful: true
```
ditto

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_develop.yaml:
```diff
@@ -0,0 +1,41 @@
+      - foreman
+    triggers:
+      - github_daily
+    axes:
```
Missing the label=fast axis still?

ehelms commented:
Updated:

 * changes for only-if-success
 * added xvfb wrapper
 * parameterized the release_nightly_build publisher
 * renamed the github trigger

ehelms commented:
I also added the label axis.

ehelms commented:
Changing this to do not merge, as I want to look at configuring this as a job-template instead of a job for easier re-use and customization.

domcleal commented:
For stable branch jobs or something else?

ehelms commented:
I was going to try to achieve better re-usability, but job-templates didn't work as I envisioned they might so no need to try to break it down further at this point. I did rename xvfb macro to default_xvfb. All in all I think this is good for final review cycle.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/triggers/github_five_hours.yaml:
```diff
@@ -0,0 +1,6 @@
+- trigger:
+    name: github_five_hours
```
five_minutes?  First column of a cron job specifier's minutes.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/triggers/github_five_hours.yaml:
```diff
@@ -0,0 +1,6 @@
+    name: github_five_hours
+    triggers:
+      - github
+      - pollscm: '*/5 * * * *'
```
H/* currently

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_develop.yaml:
```diff
@@ -0,0 +1,49 @@
+      daysToKeep: -1
+      numToKeep: 32
+    properties:
+      - github_foreman
```
Might as well put this inline, it's only really going to be applicable for this.

ehelms replied:
It would be used in other jobs such as test_develop_pr_core, test_develop_pr_rubocop but if you think it minor enough it can be inlined in all of them.

domcleal replied:
Ah yes, thanks!

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/triggers/github_five_hours.yaml:
```diff
@@ -0,0 +1,6 @@
+    triggers:
+      - github
+      - pollscm: '*/5 * * * *'
+      - schedule-failed-builds: '* * * * *'
```
This might not be useful if this scm config is to be reused, we only use it on a few jobs - generally we don't retry.

ehelms commented:
I broke the triggers out into their own individual files for better composition.

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/triggers/github_fifteen_hours.yaml:
```diff
@@ -0,0 +1,4 @@
+- trigger:
+    name: scm_fifteen_minutes
```
The filename doesn't match the name, should be "minutes".

domcleal commented on puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/test_proxy_develop.yaml:
```diff
@@ -10,9 +10,7 @@
-      - github
-      - pollscm: '*/5 * * * *'
-      - schedule-failed-builds: '* * * * *'
+      - github_five_hours
```
scm_fifteen_minutes + github I think?

ehelms commented:
Updated to fix those silly mistakes with naming in the triggers.

domcleal commented:
Thanks @ehelms!

Please just add a comment when you've rebased the other PR(s).
